ChaosMod/Effects/db/Player/PlayerBladeHunger.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+       Effect by ProfessorBiddle
+*/
+
+#include <stdafx.h>
+
+static int lastPlayerKills, fade, alpha, orange;
+static Hash currentPedHash;
+static Hash bladeHash;
+
+static void OnStart()
+{
+	currentPedHash = -1;
+	lastPlayerKills = -1;
+	bladeHash = GET_HASH_KEY("WEAPON_MACHETE");
+	fade = 0;
+	alpha = 0;
+	orange = 0;
+}
+
+static void OnTick()
+{
+	Ped playerPed = PLAYER_PED_ID();
+
+	for (Hash weapon : Memory::GetAllWeapons())
+	{
+		if (weapon != bladeHash)
+		{
+			REMOVE_WEAPON_FROM_PED(playerPed, weapon);
```

Comment:
Why remove all weapons? You can just force equip a machete and disable weapon switching

Reply:
I didn't think about that at all, that's a good idea lol

Reply:
Now add blocking any weapon-change 🤪

Reply:
No need, the equipnow tag forces the player to equip the machete at all times

Reply:
sure, but why let him? ;P yeah, but looks good to me 👍